### PR TITLE
Action Service

### PR DESCRIPTION
**Pull request by SOHAMPAL23**
# Pull Requests Requirements

- Clearly describe what changes have been made in the integration tests. Include information about the specific functionality being tested and any new tests that have been added.

Testing Scope:

- Specify which components or services have been covered in the integration tests.

## Summary
This PR introduces comprehensive Spring JUnit tests for the Action module, complete with a corresponding DBUnit XML dataset. It ensures full CRUD coverage, robust edge case handling, and valid integration with the service and DAO layers.

## Related Issue
https://github.com/DIGI-UW/OpenELIS-Global-2/issues/1913

## Other

[Add any additional information or notes here]

### CODE REVIEW

File: src/test/java/org/openelisglobal/action/action.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        executeDataSetWithStateManagement("testdata/action.xml");
+    }
+
+    @AfterEach
+    public void cleanUp() throws Exception {
+        try (Connection conn = dataSource.getConnection(); Statement stmt = conn.createStatement()) {
+            stmt.execute("TRUNCATE TABLE action RESTART IDENTITY CASCADE");
```
Comment (mozzy11) [inline src/test/java/org/openelisglobal/action/action.java:35]: can you use the existing [cleanRowsInCurrentConnection](https://github.com/DIGI-UW/OpenELIS-Global-2/blob/develop/src/test/java/org/openelisglobal/BaseWebContextSensitiveTest.java#L119) util to clean the tables ??

**Comment by mozzy11** [comment]
@SOHAMPAL23 can you fix the formatting issue ??

**Comment by SOHAMPAL23** [comment]
@mozzy11 Yeah working on it

**Comment by adityadeshlahre** [comment]
> @mozzy11 Yeah working on it

RUN `mvn spotless:apply` in main project dir and commit the changes

**Comment by mozzy11** [comment]
Thanks @SOHAMPAL23 .
Can you ensure the tests run fine in your local enviroment ??

File: src/test/java/org/openelisglobal/action/action.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import java.util.List;
+import javax.sql.DataSource;
+import org.junit.jupiter.api.*;
+import org.openelisglobal.BaseWebContextSensitiveTest;
+import org.openelisglobal.action.service.ActionService;
+import org.openelisglobal.action.valueholder.Action;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.transaction.annotation.Transactional;
+
+@Transactional
```
Comment (mherman22) [inline src/test/java/org/openelisglobal/action/action.java:15]:
```suggestion
```

**Review by mherman22 — CHANGES_REQUESTED** [review]

**Comment by mozzy11** [comment]
Thanks @SOHAMPAL23 .
This has already been implemented
